edx/frontend-app-gradebook
# Conditionally enable segment and clean up environment variables

This PR does a few related things:

- It bumps @edx/frontend-platform to latest, picking up a fix that allows apps to conditionally load Segment if SEGMENT_KEY is present
- It does some light cleanup on the .env and .env.development files, most importantly setting default values to an empty string `''` - which is falsy - rather than `null` which gets cast to `"null"` when ingested into the app.  
- It conditionalizes the loading of segmentMiddleware based on the truthiness of `configuration.SEGMENT_KEY` - this is after I realized that gradebook isn't yet uses SegmentAnalyticsService from frontend-platform, but rolls its own segment integration.  Figured the bump to frontend-platform wouldn't hurt.

Build-Test-Release working group's issue about this: https://github.com/openedx/build-test-release-wg/issues/6
CE Squad's task: https://openedx.atlassian.net/browse/CENG-53

**Developer Checklist**

**Testing Instructions**

Before this PR, a 404 request was made to: https://cdn.segment.com/analytics.js/v1/null/analytics.min.js on load when SEGMENT_KEY was set to 'null' in .env.development

After this PR, the 404 request should no longer happen, and segmentMiddleware won't be applied in situations where SEGMENT_KEY is falsy.  Which it now is in development by default.

There should be no other noticeable changes or regressions.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:



FYI: @edx/masters-devs-gta
